hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -871,7 +876,10 @@ static FSNamesystem loadFromDisk(Configuration conf) throws IOException {
     this.contextFieldSeparator =
         conf.get(HADOOP_CALLER_CONTEXT_SEPARATOR_KEY,
             HADOOP_CALLER_CONTEXT_SEPARATOR_DEFAULT);
-    fsLock = new FSNamesystemLock(conf, detailedLockHoldTimeMetrics);
+    Class<? extends FSNamesystemLock> lockKlass = conf.getClass(
+        DFS_NAMENODE_LOCK_MODEL_PROVIDER_KEY, DFS_NAMENODE_LOCK_MODEL_PROVIDER_DEFAULT,
+        FSNamesystemLock.class);
+    fsLock = createLock(lockKlass, conf, detailedLockHoldTimeMetrics);
```

Comment:
This is used for the FGL implementation, right? GlobalFSNamesystemLock is the original lock? If Im right, GlobalFSNamesystemLock should be the default class?

Reply:
Yes, `GlobalFSNamesystemLock` should be the default class. 

But in order to find dead locks for the other PRs via UT, I just change this value to FGL first.

After we complete all sub-tasks of this milestone, I will change the default value to GlobalFSNamesystemLock.

Reply:
done